### PR TITLE
`Learning analytics`: Add exercise statistics to student course dashboard

### PR DESCRIPTION
**Pull request by FelixTJDietrich**
> [!WARNING]
> 😎 

#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [ ] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Server
Server code see #8508

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [ ] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

We want to display a overview/dashboard over the student's progress in a course.

### Description

Adds exercise statistics to the student course dashboard:
- Overall performance progress bar
  - Shows the overall archived points so far out of a maximum
- Exercise performance overview
  - Shows the exercise scores of the student alongside the course average as a line chart
- Exercise lateness overview 
  - Shows the exercise lateness of the student alongside the course average as a line chart
  - Lateness is how much of the working duration you have used for an exercise (0% is handed in early, 100% is late)

### Steps for Testing

Prerequisites:
- 1 Admin (for feature flag and enabling the course dashboard)
- 1 Instructor
- 2 Students
- 1 Course

#### Turn on Student Course Dashboard feature flag and dashboard for course (if necessary)
As an Admin
1. Log in to Artemis
2. Navigate to Server Administration -> Feature Toggles 
3. Turn on StudentCourseAnalyticsDashboard
4. Navigate to Course Management 
5. Edit Course 
6. Enable "Student Course Analytics Dashboard Enabled" -> Save
7. Create an (Text-)Exercise that starts now 

#### Checkout Exercise Statistics in Course Dashboard
As a Student
1. Navigate to the course
2. Navigate to Dashboard (Above "Exercises")
3. Verify that what you see makes sense (compare with Statistics for example)
4. Participate in the created Exercise
5. Navigate to the Dashboard
6. Verify that what you see makes sense (compare with Statistics for example)

As Instructor
7. Navigate to the Assessment Dashboard of the Exercise
8. Assess the submission (might need to move the due date up by editing the exercise)
9. Submit assessment

As Student
10. Navigate to the course dashboard again
11. Verify that it still makes sense (points etc. displayed correctly)

12. Try with more exercises
13. Check for responsiveness

#### Performance Review
- [ ] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [ ] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

### Screenshots

**Student participating in some exercises**
Screenshot 2024-05-13 at 16 40 33

**Student that did not participate in any exercises**
Screenshot 2024-05-13 at 16 41 44

**No data so far**
Screenshot 2024-05-13 at 16 42 08

**No score/assessment for exercise -> 0**
Screenshot 2024-05-13 at 17 11 28

### CODE REVIEW

**Comment by FelixTJDietrich** [comment]
TODO: Client tests for the charts

**Review by kaancayli — APPROVED** [review]
Overall code looks good. Approved 👍🏻

File: src/test/java/de/tum/in/www1/artemis/MetricsIntegrationTest.java [hunk 1]
```diff
@@ -86,7 +85,6 @@ void shouldReturnExerciseInformation() throws Exception {
             assertThat(exerciseInformation).allSatisfy((id, dto) -> assertThat(id).isEqualTo(dto.id()));
         }
 
-        @Disabled // TODO: reduce jacoco missing by one after enabled
```
Comment (MaximilianAnzinger) [inline src/test/java/de/tum/in/www1/artemis/MetricsIntegrationTest.java:89]: don't disable

**Review by MaximilianAnzinger — CHANGES_REQUESTED** [review]

**Comment by FelixTJDietrich** [comment]
Idk somehow GitHub is a bit buggy right now not showing if the changes are resolved. If I click view file it is resolved.

**Pull request merged**